src/Pages/VideoListingPage/VideoListingPage.jsx

```diff
@@ -0,0 +1,106 @@
+import React, {useEffect, useState} from 'react';
+import { useParams } from 'react-router-dom';
+import { useAlert } from '../../Context';
+import { VideoCard } from '../../Shared';
+import './VideoListingPage.css';
+import { getCategoryById , getVideos, getAllCategory} from '../../Helpers/Services/actions';
+import { v4 as uuid } from "uuid";
+
+const VideoListingPage = () =>{
+
+    const [data, setData] = useState({});
+    const [ videoData, setVideoData] = useState([]);
+    const [ categoryData, setCategoryData] = useState([]);
+    const { categoryId } = useParams();   
+    const {alertContent , setAlertContent} = useAlert();
+    const [selectedCategory, setSelectedCategory] = useState('')
+
+    useEffect(()=> {
+        getCategoryById({categoryId: categoryId})
+        .then(res=> {
+            if(res.status === 200){
+                setData(res.data.category);
+            }else{
+                setAlertContent({_id: uuid(), isShow:true, type:'ERROR', content:"Unexpected error.Please try again later."})
+            }            
+        })
+        .catch((error) => {
+                setAlertContent({_id: uuid(), isShow:true, type:'ERROR', content:"Unexpected error.Please try again later."})
+            })
+        }
+    ,[categoryId])
+
+    useEffect(()=> {
+        getVideos()
+        .then(res=> {
+            if(res.status === 200){
+                setVideoData(res.data.videos);
+            }else{
+                setAlertContent({_id: uuid(), isShow:true, type:'ERROR', content:"Unexpected error.Please try again later."})
+            }            
+        })
+        .catch((error) => {
+                setAlertContent({_id: uuid(), isShow:true, type:'ERROR', content:"Unexpected error.Please try again later."})
+            })
+        }
+    ,[])
+
+    useEffect(()=> {
+        getAllCategory()
+        .then(res=> {
+            if(res.status === 200){
+                setCategoryData(res.data.categories);
+                const sel_ctgry = res.data.categories.find(ctgry => ctgry._id === categoryId);
```

Comment:
This probably can be written in the same useEffect, not sure though.

Reply:
this is done to increase the code redability